app/controlplane/internal/service/group.go

```diff
@@ -479,6 +479,69 @@ func (g *GroupService) UpdateMemberMaintainerStatus(ctx context.Context, req *pb
 	return &pb.GroupServiceUpdateMemberMaintainerStatusResponse{}, nil
 }
 
+// ListProjects retrieves a paginated list of projects that a group is a member of.
+func (g *GroupService) ListProjects(ctx context.Context, req *pb.GroupServiceListProjectsRequest) (*pb.GroupServiceListProjectsResponse, error) {
+	currentOrg, err := requireCurrentOrg(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	currentUser, err := requireCurrentUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse orgID
+	orgUUID, err := uuid.Parse(currentOrg.ID)
+	if err != nil {
+		return nil, errors.BadRequest("invalid", "invalid organization ID")
+	}
+
+	// Parse groupID and groupName from the request
+	id, name, err := req.GetGroupReference().Parse()
+	if err != nil {
+		return nil, errors.BadRequest("invalid", fmt.Sprintf("invalid group reference: %s", err.Error()))
+	}
+
+	// Parse requesterID (current user)
+	requesterUUID, err := uuid.Parse(currentUser.ID)
+	if err != nil {
+		return nil, errors.BadRequest("invalid", "invalid user ID")
+	}
+
+	// Initialize the options for getting projects
+	groupOpts := &biz.ListProjectsByGroupOpts{
```

Comment:
why aren't we loading here the list of projects the user has access to with visibleProjects as we do in other places?

I believe @jiparis mentioned smth similar in another PR, about where to add the authorization logic.

Reply:
Yes, that's what I mention in the other comment. We created that method just for this purpose. @javirln just call `visibleProjects` and you'll get the filtered list already. A `nil` list means no filter at all. So the code in Data is valid.

Reply:
I added a couple of these yesterday, this is an example https://github.com/chainloop-dev/chainloop/blob/f75ad1d449c98c8ab7eab80f4233e33387b296a0/app/controlplane/internal/service/workflowcontract.go#L51